static/src/js/containers/SavedProjectsContainer/__tests__/SavedProjectsContainer.test.jsx

```diff
@@ -0,0 +1,228 @@
+      expect(screen.getByRole('button', { name: /remove project/i })).toBeInTheDocument()
+    })
+  })
+
+  describe('When the authToken is null', () => {
+    test('renders no projects', async () => {
+      setup({
+        overrideProps: {
+          authToken: null
+        }
+      })
+
+      await waitFor(() => {
+        expect(screen.queryByRole('button', { name: /remove project/i })).toBeNull()
+      })
+    })
+  })
+
+  describe('When deleting a project', () => {
+    const projectId = '123'
+
+    test('successfully deletes the project and shows a success toast', async () => {
+      const mockProjects = [
+        {
+          id: projectId,
+          name: 'Project 1',
+          path: '/search?p=!C123',
+          created_at: '2022-01-01T00:00:00.000Z'
+        }
+      ]
+
+      nock(/localhost/)
+        .get(/projects/)
+        .reply(200, mockProjects)
+
+      nock(/localhost/)
+        .delete(`/projects/${projectId}`)
+        .reply(204)
+
+      window.confirm = jest.fn(() => true)
+
+      setup()
+
+      const removeBtn = await screen.findByRole('button', { name: /remove project/i })
+
+      // eslint-disable-next-line testing-library/no-unnecessary-act
+      await act(async () => {
+        await userEvent.click(removeBtn)
+      })
+
+      expect(addToast).toHaveBeenCalledTimes(1)
+      await waitFor(() => {
+        expect(addToast).toHaveBeenCalledWith('Project removed', {
+          appearance: 'success',
+          autoDismiss: true
+        })
+      })
+    })
+
+    test('handles an error during project deletion', async () => {
+      const mockProjects = [
+        {
+          id: projectId,
+          name: 'Project 1',
+          path: '/search?p=!C123',
+          created_at: '2022-01-01T00:00:00.000Z'
+        }
+      ]
+
+      nock(/localhost/)
+        .get(/projects/)
+        .reply(200, mockProjects)
+
+      nock(/localhost/)
+        .delete(`/projects/${projectId}`)
+        .replyWithError('Failed to delete')
+
+      window.confirm = jest.fn(() => true)
+
+      const { props } = setup()
+
+      const removeBtn = await screen.findByRole('button', { name: /remove project/i })
+      await userEvent.click(removeBtn)
+
+      await waitFor(() => {
+        expect(props.dispatchHandleError).toHaveBeenCalledTimes(1)
+      })
+
+      await waitFor(() => {
+        expect(props.dispatchHandleError).toHaveBeenCalledWith(
+          expect.objectContaining({
+            error: expect.any(Error),
+            action: 'handleDeleteSavedProject',
+            resource: 'project',
+            verb: 'deleting',
+            notificationType: 'banner'
+          })
+        )
+      })
+    })
+  })
+
+  describe('When fetching projects fails', () => {
+    test('calls dispatchHandleError with the correct error data', async () => {
+      nock(/localhost/)
+        .get(/projects/)
+        .replyWithError('Failed to fetch')
+
+      const { props } = setup()
+
+      await waitFor(() => {
```

Comment:
You don't need both of the `waitFor`. In the case where you have to wait, I find it more reliable to wait for the `toHaveBeenCalledWith` (sometimes that will have different values), then after the `waitFor` you can have the `toHaveBeenCalledTimes`.

When I removed this waitFor I got an error, but putting it after the `waitFor` wrapping `toHaveBeenCalledWith` works

Reply:
Why was this change reverted?